R/animint.R

```diff
@@ -59,6 +59,63 @@ parsePlot <- function(meta){
   ## grid 0-1 scale). This allows transformations to be used
   ## out of the box, with no additional d3 coding.
   theme.pars <- ggplot2:::plot_theme(meta$plot)
+  
+  ## extract panel backgrounds from theme.pars
+  get_bg <- function(pars) {
+    # if no border specified 
+    if(length(pars) == 0) {
+      pars <- list(fill = NULL, colour = NULL, size = NULL, linetype = NULL)
+    } else {
+      # convert fill to RGB if necessary
+      if(!(is.rgb(pars$fill))) { 
+        pars$fill <- toRGB(pars$fill)
+      }
+      # if border color is specified
+      if(!is.na(pars$colour)) {
+        # convert color to RGB if necessary
+        if(!(is.rgb(pars$colour))) { 
+          pars$colour <- toRGB(pars$colour)
+        }
+        # check if the user specified linetype for the border
+        if(is.null(pars$linetype)) {
+          pars$linetype <- "solid"
+        } else {
+          # make sure linetype is correctly specified
+          stopifnot(
+            pars$linetype %in% c("blank", "solid", "dashed", "dotted", "dotdash", "longdash", "twodash") | 
+              is.numeric(pars$linetype))
+        }
+      }
+    }
+    pars
+  }
+
+  # saving background info
+  plot.meta$panel_background <- get_bg(theme.pars$panel.background)
+  plot.meta$panel_border <- get_bg(theme.pars$panel.border)
```

Comment:
Here, I'm extracting the `panel_background` and `panel_border` separately.  I believe that `panel_background` is used to `fill` the background rectangle and `panel_border` is used to draw the border around each panel